lib/Sema/CSApply.cpp

```diff
@@ -910,6 +910,25 @@ namespace {
 
         closeExistential(ref, locator, /*force=*/openedExistential);
 
+        // If this attribute was inferred based on deprecated Swift 3 rules,
+        // complain.
```

Comment:
I'm currently doing it across module boundaries, because I really did want to get the multiple-targets-in-a-project case. I don't think we have a way to tell the difference between a module that comes from a different package (that the user doesn't care about) vs. a different target (that the user does care about).